src/codegen/mod.rs

```diff
@@ -842,12 +845,59 @@ impl<'a> Bitfield<'a> {
     }
 }
 
+fn parent_has_method(ctx: &BindgenContext,
+                     parent: &CompInfo,
+                     name: &str)
+                     -> bool {
+    parent.methods().iter().any(|method| {
+        let method_name = match *ctx.resolve_item(method.signature()).kind() {
+            ItemKind::Function(ref func) => func.name(),
+            ref otherwise => panic!("a method's signature should always be a \
+                                     item of kind ItemKind::Function, found: \
+                                     {:?}",
+                                    otherwise),
+        };
+
+        method_name == name || ctx.rust_mangle(&method_name) == name
+    })
+}
+
+fn bitfield_getter_name(ctx: &BindgenContext,
+                        parent: &CompInfo,
+                        bitfield_name: &str)
+                        -> ast::Ident {
+    let name = ctx.rust_mangle(bitfield_name);
+
+    if parent_has_method(ctx, parent, &name) {
```

Comment:
Yeah, I looked at reusing the method overloading infrastructure but deemed it not worth the effort we'd need to refactor.